engine/apps/alerts/models/user_notification_bundle.py

```diff
@@ -0,0 +1,72 @@
+from apps.base.models.user_notification_policy import validate_channel_choice
+
+
+class UserNotificationBundle(models.Model):
+    NOTIFICATION_CHANNELS_TO_BUNDLE = [
+        UserNotificationPolicy.NotificationChannel.SMS,
+    ]
+
+    user = models.ForeignKey("user_management.User", on_delete=models.CASCADE, related_name="notification_bundles")
+    important = models.BooleanField()
+    notification_channel = models.PositiveSmallIntegerField(
+        validators=[validate_channel_choice], null=True, default=None
+    )
+    last_notified_at = models.DateTimeField(default=None, null=True)
+    notification_task_id = models.CharField(max_length=100, null=True, default=None)
+    # estimated time of arrival for scheduled send_bundled_notification task
+    eta = models.DateTimeField(default=None, null=True)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["user", "important", "notification_channel"], name="unique_user_notification_bundle"
+            )
+        ]
+
+    def notified_recently(self) -> bool:
+        return (
+            timezone.now() - self.last_notified_at < timezone.timedelta(seconds=BUNDLED_NOTIFICATION_DELAY_SECONDS)
+            if self.last_notified_at
+            else False
+        )
+
+    def eta_is_valid(self) -> bool:
+        """
+        `eta` shows eta of scheduled send_bundled_notification task and should never be less than the current time
+        (with a 1 minute buffer provided).
+        `eta` is None means that there is no scheduled task.
+        """
+        if not self.eta or self.eta + timezone.timedelta(minutes=1) >= timezone.now():
+            return True
+        return False
+
+    def get_notification_eta(self) -> datetime.datetime:
+        last_notified = self.last_notified_at if self.last_notified_at else timezone.now()
+        return last_notified + timezone.timedelta(seconds=BUNDLED_NOTIFICATION_DELAY_SECONDS)
+
+    def append_notification(self, alert_group, notification_policy):
+        self.notifications.create(
+            alert_group=alert_group, notification_policy=notification_policy, alert_receive_channel=alert_group.channel
+        )
+
+    @classmethod
+    def notification_is_bundleable(cls, notification_channel):
+        return notification_channel in cls.NOTIFICATION_CHANNELS_TO_BUNDLE
+
+
+class BundledNotification(models.Model):
+    alert_group = models.ForeignKey("alerts.AlertGroup", on_delete=models.CASCADE)
+    alert_receive_channel = models.ForeignKey("alerts.AlertReceiveChannel", on_delete=models.CASCADE)
+    notification_policy = models.ForeignKey("base.UserNotificationPolicy", on_delete=models.SET_NULL, null=True)
+    notification_bundle = models.ForeignKey(
+        UserNotificationBundle, on_delete=models.CASCADE, related_name="notifications"
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+    bundle_uuid = models.CharField(max_length=100, null=True, default=None, db_index=True)
```

Comment:
I see this is _primarly_ used right now in as the second arg to [`PhoneBackend.notify_by_sms_bundle_async`](https://github.com/grafana/oncall/pull/4457/files#diff-08e0e52cc6ff4bfe4a07d2931fe3385b62cb9265ffc646675865f63a7535c91bR197), but can we just use `pk` instead of needing also `bundle_uuid`?

Reply:
The idea is that `bundle_uuid` groups a bunch of `BundledNotification`s, that were processed in the `send_bundled_notification` task (if user got flooded, it can be a hundreds of them). While we are processing these notifications, new notifications can be created and attached to the current `UserNotificationBundle` and they will have their own `bundle_uuid` and will be processed and sent in a separate task.
That means, that we can't use `UserNotificationBundle` pk, because there could be processed and unprocessed notifications attached at the same time, and we want to keep them separately. We don't want to use `BundledNotification` pks, because there could be a long list of them